### PR TITLE
Added the possibility to specify a timeout for the request

### PR DESCRIPTION
**Pull request by marcoalbarelli**

### CODE REVIEW

**Comment by marcoalbarelli** [comment]
fixes #39 and #26

File: index.js [hunk 1]
```diff
@@ -135,7 +135,7 @@ Req.prototype._send = function () {
     }
     var req = iface.request(opts);
     
-    if (req.setTimeout) req.setTimeout(Math.pow(2, 32) * 1000);
+    if (req.setTimeout) req.setTimeout( this.options.timeout ? this.options.timeout : Math.pow(2, 32) * 1000 );
```
Comment (juliangruber) [inline index.js]: this line is over 80 chars, can you refactor?

Comment (marcoalbarelli) [inline index.js]: I moved the timeout assignment out of the way

File: index.js [hunk 2]
```diff
@@ -134,8 +134,14 @@ Req.prototype._send = function () {
...
     }
     var req = iface.request(opts);
-    
-    if (req.setTimeout) req.setTimeout(Math.pow(2, 32) * 1000);
+
+    var timeout = Math.pow(2, 32) * 1000;
+
+    if(this.options.timeout){
+        timeout = this.options.timeout;
+    }
```
Comment (juliangruber) [inline index.js]: this can be written more elegantly using a ternary assignment

Comment (marcoalbarelli) [inline index.js]: Elegance is a matter of taste :)
Could you give me a suggestion on how you would find it elegant with a ternary _and_ shorter than 80 columns?
Would setting tmp variables be fine in your perspective?

Comment (marcoalbarelli) [inline index.js]: In the rest of the code the binary `||` is used for these assignment: I'd use that one if that's right for you:
`var timeout = this.options.timeout || Math.pow(2, 32) * 1000;`

Comment (juliangruber) [inline index.js]: perfect!

**Comment by marcoalbarelli** [comment]
@juliangruber is there something still to be fixed in this PR?
If not it would be very kind of you to merge it since I need this feature to come up with some testing strategy using a reliable code

BTW I discovered that nock fires up a `connect` event _before_ applying the delay, so my testing strategy is already half screwed up .... 
But half is better than completely :)

**Comment by juliangruber** [comment]
Sorry, GitHub doesn't create notifications on new commits, so I didn't see it until you commented.

Tests aren't an absolute must, but help ensure your features continues working as this module is further worked on.

There's one last thing I need from you, then we're all good to merge, yay! Can you add a small documentation addition for this feature?

**Comment by marcoalbarelli** [comment]
Done

**Comment by marcoalbarelli** [comment]
P.S.
Tell me if you need me to squash

**Comment by juliangruber** [comment]
yes please squash, then i'll merge

**Comment by marcoalbarelli** [comment]
Squashed

**Comment by juliangruber** [comment]
thanks! published as 1.3.0

**Comment by perryh** [comment]
@3rd-Eden seems fine to me:
`var timeout = this.options.timeout || Math.pow(2, 32) * 1000;`

**Comment by 3rd-Eden** [comment]
@perryh a `0` is also seen as `false` in JavaScript so if you attempt to set the timeout to `0` as seen in the snippet below:

```js
console.log(0 || Math.pow(2, 32) * 1000);
```

The outputted value will be `4294967296000` and not `0`. There for making it impossible to set `0` as value and disable the timeout.

**Comment by perryh** [comment]
Right. So if you pass in 0 the timeout is 2^32 ms, which is more or less the same as disabling any default timeout (2 minutes).

**Comment by 3rd-Eden** [comment]
Cool so once 0 becomes equal to 2^32 this won't be an issue anymore than.

**Comment by perryh** [comment]
@3rd-Eden good point, this should be fixed

**Pull request merged**